sdks/python/apache_beam/ml/inference/base_test.py

```diff
@@ -964,6 +982,117 @@ def test_child_class_without_env_vars(self):
       actual = pcoll | base.RunInference(FakeModelHandlerNoEnvVars())
       assert_that(actual, equal_to(expected), label='assert:inferences')
 
+  def test_model_manager_loads_shared_model(self):
+    mhs = {
+        'key1': FakeModelHandler(state=1),
+        'key2': FakeModelHandler(state=2),
+        'key3': FakeModelHandler(state=3)
+    }
+    mm = base._ModelManager(mh_map=mhs)
+    tag1 = mm.load('key1')
+    # Use bad_mh's load function to make sure we're actually loading the
+    # version already stored
+    bad_mh = FakeModelHandler(state=100)
+    model1 = multi_process_shared.MultiProcessShared(
+        bad_mh.load_model, tag=tag1).acquire()
+    self.assertEqual(1, model1.predict(10))
+
+    tag2 = mm.load('key2')
```

Comment:
I wonder if we might have side-effect if these tests run concurrently due to multi-process-sharing.

Reply:
We shouldn't because of the randomness introduced when we get the tag (the uuid bit); each model reference to a model will have a unique tag created for it, regardless of the key it represents. I also just tried running the suite 1000 times and it succeeded each time.

Note that this is actually an important property if we have multiple model managers operating at once (e.g. in a pipeline that does multiple sets of sequential inference)

Reply:
> Note that this is actually an important property if we have multiple model managers operating at once (e.g. in a pipeline that does multiple sets of sequential inference)

Sounds like two different model managers will not share the models, even if they are the same. Something important to keep in mind for planning memory consumption if users do sequential inferences.